### ISSUE TITLE
Component: Table

### ISSUE THREAD

**Issue by SpartanInteractive** [primefaces/primeng#13207]
### Describe the bug

The documentation of the table TableRowSelectEvent and TableRowUnSelectEvent states that the data returned is an any[].
While actually using the event it seems that object is not an array but a single object.

### Environment

We have a professional production environment using PrimeNg components for the interface.

### Angular version

16.0.4

### PrimeNG version

16.0.0-rc.2

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

v18.12.1

### Expected behavior

Either the documentation should not state it returns an array or the table should return an array.

**Comment by thetric** [comment]
Incorrectly introduced in https://github.com/primefaces/primeng/commit/f3adac59a499527d7741f63b9965fe43c8cd867f

**Comment by thetric** [comment]
The `data` field of `TableRowUnSelectEvent` is also incorrectly typed

**Comment by macedo8** [comment]
This is already fixed?

**Comment by cetincakiroglu** [comment]
Hi,

Types for the `TableRowSelectEvent` and `TableRowUnselectEvent` are as below and they don't mention the data is an array.  

Also, in the documentation data is again not an array for these event types. You can check the documentation here: https://primeng.org/table#api.table.events.TableRowSelectEvent

```
export interface TableRowSelectEvent {
    /**
     * Browser event.
     */
    originalEvent?: Event;
    /**
     * Row data.
     */
    data?: any;
    /**
     * Selection type.
     */
    type?: string;
    /**
     * Index of the selected row.
     */
    index?: number;
}
/**
 * Custom unselect event.
 * @see {@link Table.onRowUnselect}
 * @extends {TableRowSelectEvent}
 * @group Events
 */
export interface TableRowUnSelectEvent extends TableRowSelectEvent {}
```

Mentioned issue may have been fixed between the versions 16.0.0-rc.2 - 16.6.0

Closing the issue since the mentioned problem does not exist. If any error occurs, please feel free to open a new issue by providing a code example.
